### PR TITLE
SONAR-6462 Add deprecated flag on WS and WS parameters

### PR DESCRIPTION
**Pull request by teryk**

### CODE REVIEW

**Comment by SonarTech** [comment]
SonarQube analysis reported 5 issues:
* MAJOR 5 major

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR Add a 'package-info.java' file to document the 'protocol' package (org.sonarsource.sonarqube:sonar-batch-protocol:target/generated-sources/protobuf/org/sonar/batch/protocol) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1228)
* MAJOR Add a 'package-info.java' file to document the 'input' package (org.sonarsource.sonarqube:sonar-batch-protocol:target/generated-sources/protobuf/org/sonar/batch/protocol/input) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1228)
* MAJOR Add a 'package-info.java' file to document the 'output' package (org.sonarsource.sonarqube:sonar-batch-protocol:target/generated-sources/protobuf/org/sonar/batch/protocol/output) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1228)
* MAJOR Add a 'package-info.java' file to document the 'protobuf' package (org.sonarsource.sonarqube:sonar-db:target/generated-sources/protobuf/org/sonar/db/protobuf) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1228)
* MAJOR Add a 'package-info.java' file to document the 'ws' package (org.sonarsource.sonarqube:sonar-ws:target/generated-sources/protobuf/org/sonarqube/ws) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1228)

File: sonar-plugin-api/src/main/java/org/sonar/api/server/ws/WebService.java [hunk 1]
```diff
@@ -290,6 +291,11 @@ public NewAction setSince(@Nullable String s) {
       return this;
     }
 
+    public NewAction setDeprecatedSince(@Nullable String deprecatedSince) {
```
Comment (julienlancelot) [inline sonar-plugin-api/src/main/java/org/sonar/api/server/ws/WebService.java]: Why a string and not a float ?

**Comment by julienlancelot** [comment]
Some deprecated properties have been forgotten : 
- api/issues/assign?me
- api/issues/assign?componentRootUuids&componentRoots&components&projects

**Comment by SonarTech** [comment]
SonarQube analysis reported 5 issues:
* MAJOR 5 major

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR Add a 'package-info.java' file to document the 'protocol' package (org.sonarsource.sonarqube:sonar-batch-protocol:target/generated-sources/protobuf/org/sonar/batch/protocol) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1228)
* MAJOR Add a 'package-info.java' file to document the 'input' package (org.sonarsource.sonarqube:sonar-batch-protocol:target/generated-sources/protobuf/org/sonar/batch/protocol/input) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1228)
* MAJOR Add a 'package-info.java' file to document the 'output' package (org.sonarsource.sonarqube:sonar-batch-protocol:target/generated-sources/protobuf/org/sonar/batch/protocol/output) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1228)
* MAJOR Add a 'package-info.java' file to document the 'protobuf' package (org.sonarsource.sonarqube:sonar-db:target/generated-sources/protobuf/org/sonar/db/protobuf) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1228)
* MAJOR Add a 'package-info.java' file to document the 'ws' package (org.sonarsource.sonarqube:sonar-ws:target/generated-sources/protobuf/org/sonarqube/ws) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1228)

File: sonar-plugin-api/src/main/java/org/sonar/api/server/ws/WebService.java (hunk 1 above)
Comment (teryk) [inline sonar-plugin-api/src/main/java/org/sonar/api/server/ws/WebService.java]: to be consistent with the other since parameters?

Comment (julienlancelot) [inline sonar-plugin-api/src/main/java/org/sonar/api/server/ws/WebService.java]: But why ? A version should always be a double ?

Comment (teryk) [inline sonar-plugin-api/src/main/java/org/sonar/api/server/ws/WebService.java]: Imagine I want to put 5.1.2 ?

**Comment by teryk** [comment]
It's not in the scope of this ticket to add all the deprecated WS and parameters

File: sonar-plugin-api/src/main/java/org/sonar/api/server/ws/WebService.java (hunk 1 above)
Comment (julienlancelot) [inline sonar-plugin-api/src/main/java/org/sonar/api/server/ws/WebService.java]: Ok, you get the point !

**Comment by julienlancelot** [comment]
Ok, I let you create the JIRA ticket to set the deprecated flag on deprecated WS / parameters.
Otherwise LGTM.

**Comment by teryk** [comment]
https://jira.sonarsource.com/browse/SONAR-6923

**Pull request merged**